src/wp-includes/block-patterns.php

```diff
@@ -319,37 +319,132 @@ function _register_remote_theme_patterns() {
 
 /**
  * Register any patterns that the active theme may provide under its
- * `./patterns/` directory. Each pattern is defined as a PHP file and defines
- * its metadata using plugin-style headers. The minimum required definition is:
+ * `./patterns/` directory.
  *
- *     /**
- *      * Title: My Pattern
- *      * Slug: my-theme/my-pattern
- *      *
+ * @since 6.0.0
+ * @since 6.1.0 The `postTypes` property was added.
+ * @since 6.2.0 The `templateTypes` property was added.
+ * @since 6.4.0 Uses the `_wp_get_block_patterns` function.
+ * @access private
+ */
+function _register_theme_block_patterns() {
+	/*
+	 * Register patterns for the active theme. If the theme is a child theme,
+	 * let it override any patterns from the parent theme that shares the same slug.
+	 */
+	$themes   = array();
+	$theme    = wp_get_theme();
+	$themes[] = $theme;
+	if ( $theme->parent() ) {
+		$themes[] = $theme->parent();
+	}
+	$registry = WP_Block_Patterns_Registry::get_instance();
+
+	foreach ( $themes as $theme ) {
+		$pattern_data = _wp_get_block_patterns( $theme );
+		$dirpath      = $theme->get_stylesheet_directory() . '/patterns/';
+		$text_domain  = $theme->get( 'TextDomain' );
+
+		foreach ( $pattern_data['patterns'] as $file => $pattern_data ) {
```

Comment:
See my other comment: The function would make more sense if it only returned the actual patterns array.
```suggestion
		$patterns    = _wp_get_block_patterns( $theme );
		$dirpath     = $theme->get_stylesheet_directory() . '/patterns/';
		$text_domain = $theme->get( 'TextDomain' );

		foreach ( $patterns as $file => $pattern_data ) {
```